contracts/test/MockHubPool.sol

```diff
@@ -0,0 +1,86 @@
+    /// @notice Emitted when the adapter contract address is changed.
+    event AdapterChanged(address indexed oldAdapter, address indexed newAdapter);
+
+    /// @notice The address of the contract owner, set to the deployer of the contract.
+    address public immutable owner;
+    /// @notice The address of the adapter contract responsible for handling
+    ///         token relay and message passing to Layer 2.
+    address public adapter;
+
+    /// @notice Creates a new MockHubPool and sets the owner and initial adapter.
+    /// @param _adapter The address of the initial adapter contract.
+    constructor(address _adapter) {
+        adapter = _adapter;
+        owner = msg.sender; // Set the contract deployer as the owner
+    }
+
+    /// @notice Ensures that a function is callable only by the contract owner.
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Caller is not the owner");
+        _;
+    }
+
+    /// @notice Fallback function to receive ETH.
+    fallback() external payable {} // solhint-disable-line no-empty-blocks
+
+    /// @notice Receive function to handle direct ETH transfers.
+    receive() external payable {} // solhint-disable-line no-empty-blocks
+
+    /// @notice Changes the adapter contract address.
+    /// @dev This function can only be called by the contract owner.
+    /// @param _adapter The new adapter contract address.
+    function changeAdapter(address _adapter) public onlyOwner {
+        require(_adapter != address(0), "NO_ZERO_ADDRESS");
+        address _oldAdapter = adapter;
+        adapter = _adapter;
+        emit AdapterChanged(_oldAdapter, _adapter);
+    }
+
+    /// @notice Relays tokens from L1 to L2 using the adapter contract.
+    /// @dev This function delegates the call to the adapter contract.
+    /// @param l1Token The address of the L1 token to relay.
+    /// @param l2Token The address of the L2 token to receive.
+    /// @param amount The amount of tokens to relay.
+    /// @param to The address on L2 to receive the tokens.
+    function relayTokens(
+        address l1Token,
+        address l2Token,
+        uint256 amount,
+        address to
+    ) external {
+        (bool success, ) = adapter.delegatecall(
+            abi.encodeWithSignature(
+                "relayTokens(address,address,uint256,address)",
+                l1Token, // l1Token.
+                l2Token, // l2Token.
+                amount, // amount.
+                to // to. This should be the spokePool.
+            )
+        );
+        require(success, "delegatecall failed");
+    }
+
+    /// @notice Queries the balance of a user for a specific L2 token using the adapter contract.
+    /// @dev This function delegates the call to the adapter contract to execute the balance check.
+    /// @param l2Token The address of the L2 token.
+    /// @param user The user address whose balance is being queried.
+    function balanceOf(address l2Token, address user) external {
```

Comment:
why not just implement a more generalized function that takes in `bytes memory l2CallData` so that we can test with any arbitrary function? I think `balanceOf` is an interesting check but I wonder if there are additional test cases we'd want to send with state-modifyin xchain functions

Reply:
Absolutely. I wrote this test to include something related to arbitrary messaging. I later tested the messaging with a hub pool deployment. 

However, since we're including this in the repo, I'll update it to include arbitrary messaging.